test/graph_rewrites/backend_manager_test.cc

```diff
@@ -138,7 +141,7 @@ TEST(BackendManager, BackendClustering) {
 
   std::set<string> skip_these_nodes = {};
 
-  ASSERT_OK(MarkForClustering(&graph, skip_these_nodes));
+  ASSERT_OK(MarkForClustering(&graph, skip_these_nodes, "CPU"));
```

Comment:
Most of the backend tests were calling MarkForClustering and checking if the right backend (the backend that was set) was getting attached as attribute. Now MarkForClustering takes backend name as input, so the test is kind of trivial.

The optimizer pass already has backend_name as data member, if these functions can be made member functions of the rewrite pass/optimizer classes we wont need to pass these as arguments. we are incrementally having more arguments being passed around for different functions :)